integration/testrunner/db.go

```diff
@@ -16,45 +16,62 @@ package testrunner
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"testing"
 
 	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/logging/apiv2/loggingpb"
 	"github.com/sethvargo/go-retry"
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/googleapis/cloud/audit"
+	"google.golang.org/protobuf/encoding/protojson"
 )
 
-// queryIfAuditLogExists queries the DB and checks if audit log contained in the query exists or not.
-func queryIfAuditLogExists(ctx context.Context, tb testing.TB, query *bigquery.Query, expectedNum int64) (bool, error) {
+// queryAuditLogs queries the DB and checks if audit log contained in the query exists or not and return the results.
+// For retryable cases, return error. For non-retryable cases, fail now.
+func queryAuditLogs(ctx context.Context, tb testing.TB, query *bigquery.Query) ([]*loggingpb.LogEntry, error) {
 	tb.Helper()
-
 	job, err := query.Run(ctx)
 	if err != nil {
-		return false, fmt.Errorf("failed to run query: %w", err)
+		tb.Logf("failed to run query: %v", err)
```

Comment:
Like other places where you return err, I don't think you need to log error here.

Reply:
done